include/DataFile.h

```diff
@@ -58,6 +58,8 @@ class EXPORT DataFile : public QDomDocument
 
 	virtual ~DataFile();
 
+	bool validate( QString extension );
```

Comment:
Looking into this further, It is possible to create a DataFile, directly from a ByteArray of the xml. in this case the DataFile would never know the filename.